src/java/org/apache/cassandra/utils/Reducer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.utils;
+
+/** Accumulator that collects values of type A, and outputs a value of type B. */
+public abstract class Reducer<In,Out>
+{
+    /**
+     * @return true if Out is the same as In for the case of a single source iterator
+     */
+    public boolean trivialReduceIsTrivial()
```

Comment:
Could we change this to simply `reduceIsTrivial()`?

Reply:
Renamed to `singleSourceReduceIsTrivial()`.